src/mscorlib/src/System/Threading/CancellationToken.cs

```diff
@@ -186,18 +132,12 @@ private static void ActionToActionObjShunt(object obj)
         /// <returns>The <see cref="T:System.Threading.CancellationTokenRegistration"/> instance that can 
         /// be used to deregister the callback.</returns>
         /// <exception cref="T:System.ArgumentNullException"><paramref name="callback"/> is null.</exception>
-        public CancellationTokenRegistration Register(Action callback)
-        {
-            if (callback == null)
-                throw new ArgumentNullException(nameof(callback));
-
-            return Register(
-                s_ActionToActionObjShunt,
-                callback,
-                false, // useSync=false
-                true   // useExecutionContext=true
-             );
-        }
+        public CancellationTokenRegistration Register(Action callback) =>
+            Register(
+                s_actionToActionObjShunt,
+                callback ?? throw new ArgumentNullException(nameof(callback)),
```

Comment:
> Is throw expression needed?

Yes.  It's being passed here as the object state, not as the callback, and object state isn't validated as non-null because it's allowed to be null.